mvlearn/ajive/ajive.py

```diff
@@ -0,0 +1,760 @@
+            
+            # Finding the orthogonal complement to the joint matrix
+            if self.joint_rank == 0:
+                X_orthog = X
+            else:
+                X_orthog = X - np.dot(joint_scores, np.dot(joint_scores.T, X))
+
+            # estimate individual rank using sv threshold, then compute SVD
+            if self.indiv_ranks[bn] is None:
+                max_rank = min(X.shape) - self.joint_rank  # saves computation
+                U, D, V = svd_wrapper(X_orthog, max_rank)
+                rank = sum(D > self.sv_threshold_[bn])
+
+                if rank == 0:
+                    U, D, V = None, None, None
+                else:
+                    U = U[:, 0:rank]
+                    D = D[0:rank]
+                    V = V[:, 0:rank]
+
+                self.indiv_ranks[bn] = rank
+
+            # SVD on the orthogonal complement
+            else:  # if user inputs rank list for individual matrices
+                rank = self.indiv_ranks[bn]
+                if rank == 0:
+                    U, D, V = None, None, None
+                else:
+                    U, D, V = svd_wrapper(X_orthog, rank)
+            
+            # projecting X columns onto orthogonal complement of joint_scores
+
+            if self.store_full:
+                if rank == 0:
+                    I = np.zeros(shape=blocks[bn].shape)
+                else:
+                    I = np.array(np.dot(U, np.dot(np.diag(D), V.T)))
+            else:
+                I = None  # Kill I matrix to save memory
+
+            block_specific[bn]['individual'] = {'full': I,
+                                            'scores': U,
+                                            'svals': D,
+                                            'loadings': V,
+                                            'rank': rank}
+
+            # Getting the noise matrix, E
+            if self.store_full and not issparse(X):
+                E = X - (J + I)
+            else:
+                E = None
+            block_specific[bn]['noise'] = E
+
+        # save view specific estimates
+        self.blocks = {}
+
+        # Stores info for easy information checking
+        for bn in block_specific.keys():
+            self.blocks[bn] = \
+            ViewSpecificResults(joint=block_specific[bn]['joint'],
+                                                   individual=block_specific\
+                                                   [bn]['individual'],
+                                                   noise=block_specific[bn]\
+                                                   ['noise'],
+                                                   block_name=bn,
+                                                   obs_names=obs_names,
+                                                   var_names=var_names[bn],
+                                                   m=self.centers_[bn],
+                                                   shape=blocks[bn].shape)
+
```

Comment:
can you add an attribute called self.is_fit_ which is boolean and set to False in the __init__() function, but set to True when fit() is done, and then this attribute is checked in other methods that return things only if fit() has been called already? If it hasn't return some error message like "Must call fit() first."